src/components/parts/nav/Search.astro

```diff
@@ -0,0 +1,182 @@
+---
+---
+
+<div id="dummy-search-field" class="field pt-2 pr-4">
+    <p class="control has-icons-right">
+        <input aria-label="search" class="input" type="text" placeholder="Search" style="border: 0" />
+        <span class="icon is-small is-right is-clickable">
+            <i class="fas fa-magnifying-glass"></i>
+        </span>
+    </p>
+</div>
+
+<div id="search-modal" class="modal">
+    <div id="search-box" class="box" style="z-index: 1; width: 60vw">
+        <div class="field pt-2 pr-4">
+            <p class="control has-icons-right is-expanded">
+                <input id="search-field" autocomplete="off" aria-label="search" class="input is-medium" type="text" placeholder="Search" />
+                <span class="icon is-small is-right is-clickable">
+                    <i class="fas fa-magnifying-glass"></i>
+                </span>
+             </p>
+        </div>
+
+        <p id="search-loading" class="has-text-centered is-hidden">Loading...</p>
+        <p id="search-hint" class="has-text-centered">Start typing to search</p>
+        <p id="search-no-result" class="has-text-centered is-hidden">No results found</p>
+
+        <div id="search-result-container">
+        </div>
+    </div>
+    <div id="search-modal-background" class="modal-background"></div>
+</div>
+
+<!-- Dummy element to make sure the two classes don't get optimized away -->
+<div class="has-background-info has-background-link is-hidden"></div>
```

Comment:
I feel like there is a better way to go about this

Reply:
Solved in [3f3661e](https://github.com/QuiltMC/quiltmc.org/pull/200/commits/3f3661e1b8d642ebf17b76e4a93f2386581285d8)